src/sentry/notifications/platform/slack/provider.py

```diff
@@ -72,8 +72,17 @@ class SlackNotificationProvider(NotificationProvider[SlackRenderable]):
     @classmethod
     def is_available(cls, *, organization: RpcOrganizationSummary | None = None) -> bool:
         # TODO(ecosystem): Check for the integration, maybe a feature as well
+        # I currently view this as akin to a rollout or feature flag for the registry
         return False
 
     @classmethod
     def send(cls, *, target: NotificationTarget, renderable: SlackRenderable) -> None:
-        pass
+        from sentry.integrations.slack.sdk_client import SlackSdkClient
+
+        if not isinstance(target, cls.target_class):
+            raise NotificationProviderError(
+                f"Target '{target.__class__.__name__}' is not a valid dataclass for {cls.__name__}"
+            )
+
+        client = SlackSdkClient(integration_id=target.integration_id)
```

Comment:
Could we instantiate an existing Sentry integration class instead of using the Slack SDK client directly from here? One of the objectives of this project is to avoid directly using clients/API calls directly so that we can have common code paths and error handling for these requests.

On a related note, if we need an intermediate layer to do some of the extraction/construction, then let's build it.

Reply:
done via https://github.com/getsentry/sentry/pull/100083